0022-lnd-plan-of-attack.adoc

```diff
@@ -0,0 +1,448 @@
+
+==== What are the actions we need to emit on cnd's REST API?
+
+Two make the above described protocol we happen, we need to extend cnd's REST API.
+In particular, we need to add three actions:
+
+. `lnd-send-payment`
+. `lnd-add-hold-invoice`
+. `lnd-settle-invoice`
+
+I choose to prefix those with `lnd` instead of `lightning` because "hold invoices" are a client feature of Lightning and don't exist on the protocol level.
+This set of actions allows us to keep cnd's relationship with lnd read-only.
+
+===== `lnd-send-payment`
+
+To send a payment through lnd, this action needs to contain the following parameters:
+
+. recipient lnd pubkey
+. amount
+. secret-hash
+. final-cltv-delta:
+The CLTV (Check-Lock-Time-Verify) delta of the HTLC in the last hop of the payment routing.
+This value is conceptually equivalent with the on-chain expiry value (it is just relative in this case).
+
+===== `lnd-add-hold-invoice`
+
+Adds a new hold invoice to lnd.
+This action could return up to four parameters:
+
+. amount
+. secret-hash
+. expiry:
+This allows us to specify, for how long we are willing to accept a payment for this invoice.
+The default here is 3600 seconds, e.g. 1 hour.
+We should return sensible values here, depending on the roles and the protocol that is used on the other chain.
+If this timeout is too short, Bob might not be able to pay the invoice in the X-HALight case even though we would like to proceed with the swap.
+For the HALight-X case, this would allow Bob to specify a time window up until he would like Alice to start the swap.
+In this case, the timeout should not be too long, otherwise the absolute expiry of his HTLC (assuming a HAN-protocol) is not enough for a secure swap.
+. memo
+We can return a description that says something like "COMIT atomic swap {swap-id} with {peer-id}" to make identifying invoices of swaps easier for the user.
+
+===== `lnd-settle-invoice`
+
+Settles a hold invoice.
+This action needs only one parameter:
+
+. secret
+
+== Recommendation
+
+Integrating Lightning requires effort on several fronts at the same time.
+This section proposed various steps we can take to parallelize this as much as possible.
+At the same time, emphasis is put on spending our time in a "smart" way, that is, not spending too much effort on code that is already known to be replaced.
+
+I see three main areas in which progress can be made in parallel:
+
+. The e2e test suite
+. The support for Lightning inside cnd
+. The messaging protocol
+
+=== The e2e test suite
+
+To know when we are done with integrating Lightning, we should start with extending our e2e test suite with the respective tests.
+Our e2e test suite is in a limbo state at the moment, hence the first step is to get this sorted.
+
+. Make the e2e tests run in parallel: https://github.com/comit-network/comit-rs/issues/1927 (including the dependencies)
+. Setup up the test infrastructure to start an LND instance per actor and open a channel between them
```

Comment:
It would be interesting to add an edge case with `Alice -- Charlie -- Bob` channel topography.